OpenRA.Mods.Common/Activities/DonateCash.cs

```diff
@@ -42,6 +42,12 @@ protected override void OnInside(Actor self)
 
 			if (self.Owner.IsAlliedWith(self.World.RenderPlayer))
 				self.World.AddFrameEndTask(w => w.Add(new FloatingText(target.CenterPosition, target.Owner.Color.RGB, FloatingText.FormatCashTick(payload), 30)));
+
+			foreach (var host in target.TraitsImplementing<INotifyCashTransfer>())
+				host.OnCashTransfer(target, self);
+
+			foreach (var me in self.TraitsImplementing<INotifyCashTransfer>())
+				me.OnCashTransfer(target, self);
```

Comment:
They are correct but, looking at the code now I admit it looks confusing.
The confusion comes from INotifyCashTransfer(self, donor);
I followed the confention where one of the parameter is named self but for INotifyCashTransfer, self actually means "acceptor".